### ISSUE TITLE
Understanding distributed strategy loss for tpu training

### ISSUE THREAD

**Issue by rohanmuplara** [tensorflow/tensorflow#48875]
I had some questions about distribute strategy custom training. https://www.tensorflow.org/tutorials/distribute/custom_training.

"If you are using regularization losses in your model then you need to scale the loss value by number of replicas. You can do this by using the tf.nn.scale_regularization_loss function."
Why are regularization loss treated different compared to other type of losses?

"Using tf.reduce_mean is not recommended. Doing so divides the loss by actual per replica batch size which may vary step to step."
How does the batch size vary step to step per replica? Also, I am not exactly sure why we would not tf.reduce mean if the batch size changes step instead of diving by the global batch size

**Comment by tilakrayal** [comment]
@rohanmuplara ,

We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or stand alone code to reproduce the issue faced]

Thanks

**Comment by rohanmuplara** [comment]
closed
